wherehows-frontend/app/controllers/api/v2/Dataset.java

```diff
@@ -246,6 +250,46 @@ private static AclDao initAclDao() {
     return Promise.promise(() -> ok(Json.newObject().set("health", Json.toJson(health))));
   }
 
+  public static Promise<Result> getExportPolicy(String datasetUrn) {
+    final DatasetExportPolicy exportPolicy;
+    try {
+      exportPolicy = EXPORT_POLICY_DAO.getDatasetExportPolicy(datasetUrn);
+    } catch (Exception e) {
+      if (e.toString().contains("Response status 404")) {
+        return Promise.promise(() -> notFound(_EMPTY_RESPONSE));
+      }
+
+      Logger.error("Fetch export policy fail", e);
+      return Promise.promise(() -> internalServerError(errorResponse(e)));
+    }
+
+    if (exportPolicy == null) {
+      return Promise.promise(() -> notFound(_EMPTY_RESPONSE));
+    }
+
+    return Promise.promise(() -> ok(Json.newObject().set("exportPolicy", Json.toJson(exportPolicy))));
+  }
+
+  public static Promise<Result> updateExportPolicy(String datasetUrn) {
+    final String username = session("user");
+    JsonNode requestBody = request().body().asJson();
```

Comment:
Might consider move this line after the if (username) , for readability. As if ()  is closely related to username